Weatherman.py

```diff
@@ -0,0 +1,277 @@
+
+
+def make_temperature_values(file_name):
+    with open(file_name, 'r') as file:
+        csv_reader = DictReader(file)
+        date_reading = []
+        max_temperature_values = []
+        min_temperature_values = []
+        mean_humidity_values = []
+        for row in csv_reader:
+            max_temperature_values.append(row['Max TemperatureC'])
+            min_temperature_values.append(row['Min TemperatureC'])
+            mean_humidity_values.append(row[' Mean Humidity'])
+            date_reading.append(row['PKT'])
+
+    return max_temperature_values, min_temperature_values, mean_humidity_values, date_reading
+
+
+# 2nd function
+
+def calculate_results(maximum_values, maximum_date_values,
+                      minimum_values, minimum_date_values,
+                      humidity_reading, date_values_humidity):
+
+    maximum_temperature = maximum_values[calculate_maximum_temperature(maximum_values)]
+    minimum_temperature = minimum_values[calculate_minimum_temperature(minimum_values)]
+    maximum_humidity = humidity_reading[calculate_maximum_temperature(humidity_reading)]
+
+    maximum_temperature_date = maximum_date_values[calculate_maximum_temperature(maximum_values)]
+    minimum_date_value = minimum_date_values[calculate_maximum_temperature(minimum_values)]
+    mean_humidity_date = date_values_humidity[calculate_maximum_temperature(humidity_reading)]
+
+    display_results(maximum_temperature, maximum_temperature_date,
+                    minimum_temperature, minimum_date_value,
+                    maximum_humidity, mean_humidity_date)
+
+
+def calculate_maximum_temperature(temperature_values):
+    maximum1 = 0
+    maximum2 = 0
+    for iterator in range(0, len(temperature_values)):
+        if temperature_values[iterator] != '':
+            if float(temperature_values[iterator]) >= maximum1:
+                maximum1 = float(temperature_values[iterator])
+                maximum2 = iterator
+            else:
+                temperature_values[iterator] = temperature_values[iterator]
+    return maximum2
+
+
+def calculate_minimum_temperature(temperature_values):
+    minimum1 = float(temperature_values[0])
+    minimum2 = 0
+    for iterator in range(1, len(temperature_values)):
+        if temperature_values[iterator] != '':
+            if float(temperature_values[iterator]) < minimum1:
+                minimum1 = float(temperature_values[iterator])
+                minimum2 = iterator
+            else:
+                minimum1 = minimum1
+    return minimum2
+
+
+def display_results(maximum_temperature, maximum_date, minimum_temperature, minimum_date,
+                    mean_humidity, mean_humidity_date):
+    print(f'Max temperature is {maximum_temperature}C on 'f'{change_date_format(maximum_date)}')
+    print(f'Min temperature is {minimum_temperature}C on 'f'{change_date_format(minimum_date)}')
+    print(f'Max Humidity is {mean_humidity}% on 'f'{change_date_format(mean_humidity_date)}')
```

Comment:
Look into `fstrings` carefully, we can use multiple variables in f-string like this:
```
some_str = f`hello {some_var} and hello to {another_var}`
```